test/unit/services/validation.service.test.ts

```diff
@@ -129,4 +172,80 @@ describe('ValidationService', () => {
       }
     });
   });
+
+  describe('validateDocument() with external URLs', () => {
+    let originalConfig: any;
+    const CONFIG_DIR = path.join(os.homedir(), '.asyncapi');
+    const CONFIG_FILE = path.join(CONFIG_DIR, 'config.json');
+
+    beforeEach(async () => {
+      // Backup original config
+      try {
+        const content = await fs.readFile(CONFIG_FILE, 'utf8');
+        originalConfig = JSON.parse(content);
+      } catch (err) {
+        originalConfig = null;
+      }
+    });
+
+    afterEach(async () => {
+      // Restore original config
+      if (originalConfig) {
+        await fs.writeFile(CONFIG_FILE, JSON.stringify(originalConfig, null, 2), 'utf8');
+      } else {
+        try {
+          await fs.unlink(CONFIG_FILE);
+        } catch (err) {
+          // File doesn't exist, ignore
+        }
+      }
+    });
+
+    it('should validate document with private GitHub reference', async () => {
```

Comment:
This is testing that the document fails to validate because the $ref is in a private URL and we haven't configured how to resolve it. IMHO, the name of the test is not reflecting what this is testing.

Reply:
I suggest something like the following:

```suggestion
    it('should fail to validate document with private GitHub reference when not properly configured', async () => {
```